ops/manifests/manifest.py

```diff
@@ -208,25 +208,42 @@ def create_crd(rsc):
 
     @lru_cache()
     def _safe_load(self, filepath: Path) -> List[Mapping]:
-        """Read manifest file and parse its content into dicts.
+        """Read manifest file and parse its content into list of dicts.
 
-        Lightkube can't properly read manifest files which contain List kinds.
+        Note: Lightkube can't properly read kind=List (a list of kubernetes resources).
+        Therefore this method will also flatten all kind=List resources into a python
+        list of resources.
         """
 
-        def _flatten(raw_resources):
+        def _flatten(raw_resources: Iterator) -> List[Mapping]:
+            """Flatten objects in a given python iterator which are kind=List."""
             resources = []
             for rsc in raw_resources:
-                if not isinstance(rsc, dict):
+                if not isinstance(rsc, Mapping):
                     # found a non-dict item?  Let's log it
-                    log.warning(f"Ignoring non-dictionary resource rsc='{rsc}'")
-                elif rsc.get("kind") == "List":
+                    log.warning(
+                        f"Ignoring non-dictionary resource rsc='{rsc}' in {filepath}"
+                    )
+                elif not rsc.get("kind") or not rsc.get("apiVersion"):
+                    log.warning(
+                        f"Ignoring non-kubernetes resource rsc='{rsc}' in {filepath}"
+                    )
```

Comment:
I can imagine a worst case scenario where `rsc` is a Secret that's missing `apiVersion` or `kind`, in which case we would accidentally log secret data here.

We're not generally loading secrets via this method right? Seems like those generally fall outside the realm of ops-lib-manifest? So maybe it's not a concern worth addressing.

Reply:
I'm not sure i understand a `Secret` that doesn't have an `apiVersion` or `kind`

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: mysecret
type: Opaque
data:
  username: YWRtaW4=
  password: YmlyZHNhcmVudHJlYWw=
```

Reply:
Yeah, it'd be a really extreme corner case where somebody rendered a file that's *supposed* to be a Kubernetes secret resource, but then forgot to include the `apiVersion` or `kind` field when rendering it. Then if that flows through here, it'll get logged as an invalid resource, but the secret data would appear in the logs.

Again, seems pretty unlikely and it's probably not an issue worth addressing.

Reply:
right, in this case i'd really not expect this case.  usually the manfiests are flat file checked into source -- not files that require to be first rendered.

this library hasn't taken extreme care to ensure any data hasn't leaked yet -- likely should be a bug on the library i'll go add right now.